src/clr-core/common/utils/register.ts

```diff
@@ -14,12 +15,14 @@ const addElementToRegistry = curryN(
       console.warn(`${tagName} has already been registered`);
     } else {
       registry.define(tagName, elementClass);
+      window.CDS._loadedElements.push(tagName);
     }
   }
 );
 
 export function registerElementSafely(tagName: string, elementClass: any) {
   if (isBrowser() && existsInWindow(['customElements'])) {
+    setupCDSGlobal();
```

Comment:
I'm wondering if it makes more sense to move `setupCDSGlobal()` here into `addElementToRegistry()`.

This function doesn't need CDSGlobal at all. And it's just state that's spread out a bit too much for my comfort zone.